servicetalk-http-api/src/test/java/io/servicetalk/http/api/Uri3986Test.java

```diff
@@ -162,8 +166,24 @@ void emptyUserInfo() {
     }
 
     @Test
-    void schemeParsedCaseSensative() {
-        verifyUri3986("HTTP://test.com", "HTTP", null, "test.com", -1, "", "", null, null, null);
+    void schemeParsedCaseInsensitive() {
+        verifyUri3986("HTTP://test.com", "http", null, "test.com", -1, "", "", null, null, null);
+        verifyUri3986("HTTPS://test.com", "https", null, "test.com", -1, "", "", null, null, null);
+        verifyUri3986("ANY://test.com", "any", null, "test.com", -1, "", "", null, null, null);
+    }
+
+    @Test
+    void reuseWellKnownScheme() {
+        String http1 = new Uri3986("http://test.com").scheme();
+        String http2 = new Uri3986("HTTP://test.com").scheme();
+        assertThat("different http", http1, sameInstance(http2));
+
+        String https1 = new Uri3986("https://test.com").scheme();
+        String https2 = new Uri3986("HTTPS://test.com").scheme();
+        assertThat("different https", https1, sameInstance(https2));
+
+        String httpss = new Uri3986("HTTPSS://test.com").scheme();
+        assertThat("same https and httpss", https1, not(sameInstance(httpss)));
```

Comment:
It's expected because `https` != `httpss`. Doesn't look like there is much value in the last 2 lines of this test, consider verifying only `http` and `https`.